compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/todo.memoize-loops-that-produce-memoizeable-values.js

```diff
@@ -0,0 +1,16 @@
+function useHook(nodeID, condition) {
+  const graph = useContext(GraphContext);
+  const node = nodeID != null ? graph[nodeID] : null;
+
+  // (2) Instead we can create a scope around the loop since the loop produces an escapinng value
+  let value;
+  for (const key of Object.keys(node?.fields ?? {})) {
+    if (condition) {
+      // (1) We currently create a scope just for this instruction, then later prune the scope because
```

Comment:
Ahh that makes sense. I guess loops are kind of the only case that we might want to lengthen instead of completely pruning scopes. Specifically, we should rewrite / lengthen a scope if it is the outermost one in a loop.

Since flattenReactiveLoops happens before other relevant passes (flatten hook calls, propagate scope deps), this seems pretty straightforward.

Nice find!